reana_server/reana_admin/consumer.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2022 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA Admin MessageConsumer."""
+
+import json
+from typing import List
+
+import click
+
+from reana_commons.consumer import BaseConsumer
+
+
+class MessageConsumer(BaseConsumer):
+    """Consumer responsible for cleaning queues."""
+
+    def __init__(
+        self,
+        queue_name: str,
+        message_key: str,
+        messages_to_delete: List[str],
+        is_interactive: bool = False,
+        **kwargs,
+    ):
+        """Initialise the class."""
+        super(MessageConsumer, self).__init__(queue=queue_name, **kwargs)
+        self.message_key = message_key
+        self.messages_to_delete = messages_to_delete
+        self.is_interactive = is_interactive
+
+        click.secho(
+            "MessageConsumer initialized with the following settings: \n"
+            f"  - queue_name: {queue_name}\n"
+            f"  - message_key: {self.message_key}\n"
+            f"  - keys_to_delete: {self.messages_to_delete}\n"
+            f"  - interactive: {self.is_interactive}"
+        )
+
+    def get_consumers(self, Consumer, channel):
+        """Implement providing kombu.Consumers with queues/callbacks."""
+        return [
+            Consumer(
+                queues=self.queue,
+                callbacks=[self.on_message],
+                accept=[self.message_default_format],
+                prefetch_count=50,
+            )
+        ]
+
+    def on_consume_ready(self, connection, channel, consumers, **kwargs):  # noqa: D102
+        click.secho(f"Starting consuming the {self.queue.name} queue", fg="green")
+        click.secho("If you want to stop the consumer, use Ctrl+C.\n", fg="yellow")
+
+    def on_consume_end(self, connection, channel):  # noqa: D102
+        click.secho(f"Finished consuming the {self.queue.name} queue", fg="green")
+
+    def on_message(self, body, message):
+        """Remove selected messages."""
+        msg_body = json.loads(body)
+        click.secho(
+            f"New message received: \n{json.dumps(msg_body, sort_keys=True, indent=4)}"
+        )
+
+        should_remove_message = False
+
+        if self.is_interactive:
+            answer = input(
+                "Delete the above message? (Y - Yes, N - No, S - Stop consuming): "
+            )
+            if answer.startswith("Y"):
+                should_remove_message = True
+            elif answer.startswith("S"):
+                self.should_stop = True
```

Comment:
We should probably improve this a little bit: right now, if a user type a lowercase `s` or `y`, the answer is simply ignored and considered as a `No`. In particular:
1. If the command receives an invalid answer, it should ask again to the user what to do
2. It should accept both lowercase and uppercase answers
3. It should also accept `yes`/`no`/`stop` as an answer

One way to do it: `answer.lower() in ["yes", "y"]`

What do you think?

Reply:
Sounds better for UX. I will change it to what you suggested.

Reply:
Done